src/screens/common/create-edit-events/ContactDetails/ContactDetails.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback } from "react";
+import { View } from "react-native";
+import { useStyles } from "./styles";
+import { PageHeader } from "@app/components/common/PageHeader";
+import { useNavigation } from "@react-navigation/native";
+import Card from "@app/components/ui/Card";
+import { FormInput } from "@app/components/forms";
+import { useForm } from "react-hook-form";
+import Button from "@app/components/ui/Button";
+import { regexValidator } from "@app/utils/validators";
+import { getErrorMessage } from "@app/components/forms";
+import { ContactDetailsReduxProps } from "./container";
+
+type ContactDetailsProps = ContactDetailsReduxProps;
+
+const ContactDetails: React.FC<ContactDetailsProps> = ({
+  pageTitle,
+  createData,
+  updateCreateData,
+}) => {
+  const styles = useStyles();
+  const navigation = useNavigation();
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const onBack = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  const onSubmit = useCallback(
+    async (data: Record<string, any>) => {
```

Comment:
me confused, so is data being renamed to "Record" and record has a type of both "string" and "any" ?

Reply:
data is of type Record<string, any>. Renaming only happens with the colon syntax during deconstruction.